content/cucumber/step-definitions.md

```diff
@@ -253,3 +253,55 @@ Given("I have {int} {color} balls", function (int1, color) {
 });
 ```
 {{% /block %}}
+
+# Grouping step definitions
+
+Avoid writing similar step definitions as they can lead to clutter. While documenting your steps helps, making use of **helper methods** to abstract them can do wonders.
+
+For example, take the following steps :
+
+```
+    Given I go to the home page
+    Given I check the about page of the website 
+    Given I get the contact details
+```
+
+If all of these steps just open the respective webpages, you might be writing *redundant steps*. While the underlying code for these steps could be different, their **behaviour** is essentially the same, ie. *to open the Home, About or Contact page*.
+
+As such, you can use abstract helper methods to reduce them into a single step:
+
+    Given I go to the {} page
+
+with the step definition :
+
+```java
+@Given("I want to open the {string} page")
+public void i_want_to_open_page(String webpage) {
+  webpageFactory.openPage(webpage);
+}
+```
+
+```javascript
+ //TODO
+```
+
+```ruby
+ //TODO
+```
+
+```kotlin
+ //TODO
+```
+
+Your step definitions are the glue to the actual code (in this example, a factory method to decide which page to open).
+
+This helps in a number of ways,
+
+1. Less and more easily maintainable steps.
+2. Increased Scalablility: Adding tests for a new functionality with the same underlying *behaviour* is easier.
+3. Concise and easy to understand tests.
```

Comment:
Do you think it'd be better to put these in a single sentence instead?

Reply:
Use bullit points instead? And make them more concise. Eg:
```
* Better maintainability of steps
* Easier reuse of steps
* Easier to understand
```

Reply:
Sure